smart_open/s3.py

```diff
@@ -700,20 +740,181 @@ def to_boto3(self, resource):
     #
     # Internal methods.
     #
-    def _read_from_buffer(self, size=-1):
-        """Remove at most size bytes from our buffer and return them."""
-        size = size if size >= 0 else len(self._buffer)
-        part = self._buffer.read(size)
-        self._current_pos += len(part)
-        return part
 
     def _fill_buffer(self, size=-1):
-        size = max(size, self._buffer._chunk_size)
-        while len(self._buffer) < size and not self._eof:
-            bytes_read = self._buffer.fill(self._raw_reader)
-            if bytes_read == 0:
-                logger.debug('%s: reached EOF while filling buffer', self)
-                self._eof = True
+        # check if existing body range is sufficient to satisfy this request, if not close it so that it re-opens
+        # with an appropriate range.
+        is_stream_limit_before_eof = self._body is not None \
```

Comment:
Please use all() to simplify this expression.

```
is_stream_limit_before_eof = all(
    self._body is not None,
    size < 0,
    ...
)
```

this avoids the repetitive "and" and artificial line breaks.